### PR TITLE
[TT-549] Document how to write a spot linter

### PR DESCRIPTION
**Pull request by medric**
## Description, Motivation, and Context

- Why is this change required?

Moving https://docs.google.com/document/d/1pfX_4hcETMG7LvFIiPK34fxh7RrIsoKrbUSlGqPwSRI/edit# to a Readme so it can be accessed easily by developers who aim to write a new linting rule. 

- What does it do?

## Checklist:

- [ ] I've added/updated tests to cover my changes
- [x] I've created an issue associated with this PR

### CODE REVIEW

File: lib/src/linting/README.MD [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# Linting
+
+Spot linters are a set of functions whose goal is to ensure a contract definition conforms with our API guidelines. In other words, linters check whether or not a contract is semantically correct.
```
Comment (driesg) [inline lib/src/linting/README.MD]:
```suggestion
Spot linters are a set of functions whose goal is to ensure a contract definition conforms with our API guidelines.
```
I'm not sure we should talk about `semantically correct`. Linters could just describe an API guideline that is just representing a standard at Airtasker. Which at any point in time could be a workaround for a certain situation

File: lib/src/linting/README.MD [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+Linting rules can be found in files inside the spot directory (under [lib/src/linting](/lib/src/linting). Each file corresponds to a specific rule and holds the logic to validate a subset of a contract (e.g. headers definition, request/response types, etc).
+
+## Anatomy of a linting rule
+
+Let's take an example of a linting rule to demonstrate how we can catch semantic errors in a given contract  :
```
Comment (driesg) [inline lib/src/linting/README.MD]:
```suggestion
Let's take an example of a linting rule to demonstrate how we can catch errors in a given contract:
```

File: lib/src/linting/README.MD [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+To run the linter against a contract, simply do
+
+```
+$ ./bin/run lint [contract].ts
+```
+
+## Test a linting rule
+
+Testing a linting rule is necessary. It allows to ensure it works as expected and comply with the guidelines.
```
Comment (driesg) [inline lib/src/linting/README.MD]:
```suggestion
Testing a linting rule is necessary. It makes sure everything works as expected and complies with the guidelines.
```

File: lib/src/linting/README.MD [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+Every linting rule comes with a specification file. Writing tests for a specific rule consists in creating contracts inputs and programmatically call the function associated with this rule.
```
Comment (driesg) [inline lib/src/linting/README.MD]:
```suggestion
Every linting rule comes with a specification file. Writing tests for a specific rule consists in creating contracts inputs and programmatically calling the function associated with this rule.
```

**Review by driesg — APPROVED** [review]
couple of suggestions and then good to go

**Review by lfportal — APPROVED** [review]

**Pull request merged**